### ISSUE TITLE
Session user is empty when using OIDC provider

### ISSUE THREAD

**Issue by ooraini** [oauth2-proxy/oauth2-proxy#1793]
The session user is always empty when using the OIDC provider.
Provider struct:
https://github.com/oauth2-proxy/oauth2-proxy/blob/965fab422d6cd30411a6ee85a34532b276284289/providers/providers.go#L21

The `UserClaim` is never populated. The field is only used by the function `setProviderDefaults` which is not called by the OIDC provider. This results in a session data with an empty user.

I would prefer if it was configurable as the `sub` claim is usually a UUID or a random string. 

The following comment in `provder_data.go`:
```
const (
	// This is not exported as it's not currently user configurable
	oidcUserClaim = "sub"
)
```

**Comment by ooraini** [comment]
If anyone is interested, I ended up reimplementing a large set of functionality in my own project [https://github.com/ooraini/oidc-auth](https://github.com/ooraini/oidc-auth).

**Comment by josemolinai-bcs** [comment]
This incident is addressed by commit fd2807c091686a2a23f41123e3470a1076e877a0
 Fix uninitialized user claim (#1873)
